src/crypto/generichash/digest.rs

```diff
@@ -26,7 +26,7 @@ impl PartialEq for Digest {
         if other.len != self.len {
```

Comment:
BTW are you sure eq should be O(1) time?

Reply:
Same check is also done in the memcmp wrapper 
https://github.com/sodiumoxide/sodiumoxide/blob/master/src/utils.rs#L21

Reply:
Can you use the memcmp wrapper instead?

Reply:
Then `if other.len != self.len {` in this eq operator is redundant.